gap/digraph.gi

```diff
@@ -1368,6 +1423,259 @@ function(filt, n, p)
   return DigraphNC(IsMutableDigraph, RANDOM_DIGRAPH(n, Int(p * 10000)));
 end);
 
+BindGlobal("DIGRAPHS_FillOutGraph", function(n, p, adjacencyList)
+
+    local i, j;
+
+    for i in [1 .. n] do
+        for j in [1 .. n] do
+            if (not (j in adjacencyList[i])) then
+                 if Float(Random([0 .. 99])/100) < Float(p) then
+                    Add(adjacencyList[i], j);
+                 fi;
+            fi;
+        od;
+    od;
+
+    return adjacencyList;
+
+end);
+
+InstallMethod(RandomDigraphCons,
+"for IsHamiltonianDigraph, a positive integer, and a float",
+[IsHamiltonianDigraph, IsPosInt, IsFloat],
+function(filt, n, p)
+    local adjacencyList, vertices, startVertex, hamiltonianCycle, x, i, j, graph;
+  
+    adjacencyList := [];
```

Comment:
Instead of `[]`, you should use `EmptyPlist(n)`, since we know that we're going to add `n` elements to the list.  This seems weird, but it tells GAP to allocate the right amount of memory and avoids waste.